### PR TITLE
(MODULES-6745) Add testmode to tests

### PR DESCRIPTION
**Pull request by glennsarti**
This commit adds the necessary entries to the acceptance setup in order
for testmode switcher to work

This commit updates all host execution methods to execute_manifest_on methods.
Where applicable, it changes explicit exitcode calls to catch_failures or
catch_changes.

### CODE REVIEW

**Comment by glennsarti** [comment]
Ran through adhoc ok.  Manually tested with beaker in agent only mode.

File: spec/spec_helper_acceptance.rb [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
+proj_root = File.expand_path(File.join(File.dirname(__FILE__), '..'))
+staging = { module_name: 'puppetlabs-reboot' }
+local = { module_name: 'reboot', source: proj_root }
 
-  hosts.each do |host|
-    step 'Install Reboot Module Dependencies'
-    on(host, puppet('module install puppetlabs-stdlib'))
-    on(host, puppet('module install puppetlabs-registry'))
+hosts.each do |host|
+  step 'Install Reboot Module Dependencies'
```
Comment (RandomNoun7) [inline spec/spec_helper_acceptance.rb]: Do you still need these step lines if the `test_name` line was removed?

File: spec/spec_helper_acceptance.rb [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
+  on(host, puppet('module install puppetlabs-stdlib'))
+  on(host, puppet('module install puppetlabs-registry'))
 
-    step 'Install Reboot Module'
-    # in CI allow install from staging forge, otherwise from local
-    install_dev_puppet_module_on(host, options[:forge_host] ? staging : local)
-  end
+  step 'Install Reboot Module'
```
Comment (RandomNoun7) [inline spec/spec_helper_acceptance.rb]: Same as above. If we aren't in the context of a test, do we still need this?

Comment (glennsarti) [reply spec/spec_helper_acceptance.rb]: Fixed

File: spec/spec_helper_acceptance.rb (hunk 1 above)
Comment (glennsarti) [reply spec/spec_helper_acceptance.rb]: Fixed

**Pull request merged**